src/audits/DataTableHeadersMissing.js

```diff
@@ -0,0 +1,81 @@
+// Copyright 2015 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+goog.require('axs.AuditRules');
+goog.require('axs.browserUtils');
+goog.require('axs.constants.Severity');
+
+(function () {
+    function tableDoesNotHaveHeaderRow(rows) {
+        var headerRow = rows[0];
+
+        var headerCells = headerRow.children;
+
+        for (var i = 0; i < headerCells.length; i++) {
+            if (headerCells[i].tagName != 'TH') {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    function tableDoesNotHaveHeaderColumn(rows) {
+        for (var i = 0; i < rows.length; i++) {
+            if (rows[i].children[0].tagName != 'TH') {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    function tableDoesNotHaveGridLayout(rows) {
+        var headerCells = rows[0].children;
+
+        for (var i = 1; i < headerCells.length; i++) {
+            if (headerCells[i].tagName != 'TH') {
+                return true;
+            }
+        }
+
+        for (var i = 1; i < rows.length; i++) {
+            if (rows[i].children[0].tagName != 'TH') {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    axs.AuditRules.addRule({
+        name: 'dataTableHeadersMissing',
+        heading: 'Data tables should have appropriate headers',
+        url: 'https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_table_01',
+        severity: axs.constants.Severity.SEVERE,
+        relevantElementMatcher: function (element) {
+            return axs.browserUtils.matchSelector(element, 'table');
```

Comment:
Upon revisiting, remembered that this rule was based on the [WebAim interpretation of 508g](http://webaim.org/standards/508/checklist) which says that presentation tables should not have any headers at all. We updated the wiki explanation and the audit name to clarify this.